src/core/browser/qgsbrowserproxymodel.cpp

```diff
@@ -124,19 +123,19 @@ bool QgsBrowserProxyModel::filterAcceptsString( const QString &value ) const
     case Normal:
     case Wildcards:
     {
-      for ( const QRegExp &rx : mREList )
+      for ( const QRegularExpression &rx : mREList )
       {
-        if ( rx.exactMatch( value ) )
+        if ( rx.match( value ).hasMatch() )
```

Comment:
Wouldn't this need to use anchored patterns?

Reply:
@nyalldawson , as far as I got to understand things, QRegExp's exactMatch when set to QRegExp::Wildcard pattern syntax is the same as QRegularExpression( QRegularExpression::wildcardToRegularExpression( string ) ), which I'm using above.

Reply:
Then the issue comes in the changes on line 135-137, because those were previously not using an exact match. I think the only solution here is to split mREList into two separate lists, one for anchored and one for non-anchored regexs

Reply:
@nyalldawson , hm, but if the filter type is RegularExpression, mREList is populated with non anchored regular expressions (see changed line 110 above). Or am I missing something here?

Reply:
Hm, you're correct. But then... this switch here has no impact at all, since the code in the switches are all identical. Can you fix this?

Reply:
Thanks, that looks better to me!